src/H5Fint.c

```diff
@@ -1951,14 +1982,19 @@ H5F_open(const char *name, unsigned flags, hid_t fcpl_id, hid_t fapl_id)
     shared = file->shared;
     lf     = shared->lf;
 
-    /* Set the file locking flag. If the file is already open, the file
+    /* Set the file locking flags. If the file is already open, the file
      * requested file locking flag must match that of the open file.
      */
-    if (shared->nrefs == 1)
-        file->shared->use_file_locking = use_file_locking;
-    else if (shared->nrefs > 1)
+    if (shared->nrefs == 1) {
+        file->shared->use_file_locking      = use_file_locking;
+        file->shared->ignore_disabled_locks = ignore_disabled_locks;
+    }
+    else if (shared->nrefs > 1) {
         if (file->shared->use_file_locking != use_file_locking)
             HGOTO_ERROR(H5E_FILE, H5E_CANTINIT, NULL, "file locking flag values don't match");
+        if (file->shared->use_file_locking && (file->shared->ignore_disabled_locks != ignore_disabled_locks))
```

Comment:
I wasn't quite sure on this addition, but it seems reasonable that, similar to how we check the file locking setting, we would also want to check that a file which is opened multiple times has the same "ignore disabled file locks" setting each time.